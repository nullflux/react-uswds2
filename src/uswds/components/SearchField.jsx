import React from 'react';
import PropTypes from 'prop-types';

//  Provides a standard Search Field.
//
//  The search field can be one of three sizes by setting the `size` prop. The
//  valid values for the `size` prop are as follows:
//
//    - SearchField.SIZE_BIG
//    - SearchField.SIZE_MEDIUM (default)
//    - SearchField.SIZE_SMALL
//
//  For more information on when to use each of the different sizes, please
//  consult the US Web Design Standards (https://standards.usa.gov/search-bar/).

export default class SearchField extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    size: PropTypes.oneOf(['', 'big', 'small'])
  };

  static defaultProps = {
    size: ''
  };

  render() {
    let buttonContents;
    if (this.props.size === 'small') {
      buttonContents = <span className='usa-sr-only'>Search</span>;
    } else {
      buttonContents = <span className='usa-search__submit-text'>Search</span>;
    }

    return (
      <form
        className={`usa-search ${
          this.props.size ? 'usa-search--' + this.props.size : ''
        }`}
      >
        <div role='search'>
          <label className='usa-sr-only' htmlFor={this.props.id}>
            Search
          </label>
          <input
            type='search'
            className='usa-input'
            name={this.props.id}
            id={this.props.id}
          />
          <button className='usa-button' type='submit'>
            {buttonContents}
          </button>
        </div>
      </form>
    );
  }
}
