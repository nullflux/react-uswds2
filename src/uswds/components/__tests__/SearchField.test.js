import React from 'react';
import { shallow } from 'enzyme';
import SearchField from '../SearchField.jsx';

jest.unmock('../SearchField.jsx');

describe('SearchField', () => {
  it('should render correctly', () => {
    const component = shallow(
      <SearchField id="testId" />
    );
    expect(component).toMatchSnapshot();
  });

  it('is has a big variant', () => {
    let searchField = shallow(<SearchField id="testId" size="big" />);
    expect(searchField.hasClass('usa-search--big')).toBe(true);
  });

  it('is has a small variant', () => {
    let searchField = shallow(<SearchField id="testId" size="small" />);
    expect(searchField.hasClass('usa-search--small')).toBe(true);
  });

});