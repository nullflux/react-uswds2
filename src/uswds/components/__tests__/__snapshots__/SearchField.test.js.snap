// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
<form
  className="usa-search "
>
  <div
    role="search"
  >
    <label
      class="usa-sr-only"
      htmlFor="testId"
    >
      Search
    </label>
    <input
      class="usa-input"
      id="testId"
      name="testId"
      type="search"
    />
    <button
      class="usa-button"
      type="submit"
    >
      <span
        className="usa-search__submit-text"
      >
        Search
      </span>
    </button>
  </div>
</form>
`;
